src/global/utils/auth.tsx

```diff
@@ -16,86 +16,105 @@
-	logOut: () => void;
+  egoJwt: string;
+  setEgoJwt: Dispatch<SetStateAction<string>>;
+  authLoading: boolean;
+  setAuthLoading: Dispatch<SetStateAction<boolean>>;
+  logIn: (newToken: string) => void;
+  logOut: () => void;
+  permissions: string[];
+  TokenUtils: ReturnType<typeof createEgoUtils>;
 };
 
 const AuthContext = createContext<AuthContextValue>({
-	egoJwt: '',
-	setEgoJwt: () => '',
-	authLoading: false,
-	setAuthLoading: () => false,
-	logIn: () => undefined,
-	logOut: () => undefined,
+  egoJwt: "",
+  setEgoJwt: () => "",
+  authLoading: false,
+  setAuthLoading: () => false,
+  logIn: () => undefined,
+  logOut: () => undefined,
+  permissions: [],
+  TokenUtils: createEgoUtils(""),
 });
 
 export const getStoredToken = () => Cookies.get(EGO_JWT_KEY);
 
 export const storeToken = (egoToken: string) => {
-	Cookies.set(EGO_JWT_KEY, egoToken);
+  Cookies.set(EGO_JWT_KEY, egoToken);
 };
 
 export function AuthProvider({ children }: { children: ReactNode }) {
-	const storedToken = getStoredToken();
-	const [egoJwt, setEgoJwt] = useState(storedToken || '');
-	const router = useRouter();
-	const path = usePathname();
-	const loginStateOnPageLoad = path === '/logging-in' && !egoJwt.length;
-	const [authLoading, setAuthLoading] = useState(loginStateOnPageLoad);
+  const config = useAppConfigContext();
+  const storedToken = getStoredToken();
+  const [egoJwt, setEgoJwt] = useState(storedToken || "");
+  const TokenUtils = createEgoUtils(config.EGO_PUBLIC_KEY);
+  const router = useRouter();
+  const path = usePathname();
+  const loginStateOnPageLoad = path === "/logging-in" && !egoJwt.length;
+  const [authLoading, setAuthLoading] = useState(loginStateOnPageLoad);
 
-	const logIn = (newToken: string) => {
-		storeToken(newToken);
-		setEgoJwt(newToken);
-		setAuthLoading(false);
-	};
+  const logIn = (newToken: string) => {
+    storeToken(newToken);
+    setEgoJwt(newToken);
+    setAuthLoading(false);
+  };
 
-	const logOut = () => {
-		setAuthLoading(true);
-		setEgoJwt('');
-		Cookies.remove(EGO_JWT_KEY);
-		Cookies.remove(LOGIN_NONCE);
-		router.push('/');
-		setAuthLoading(false);
-	};
+  const logOut = () => {
+    setAuthLoading(true);
+    setEgoJwt("");
+    Cookies.remove(EGO_JWT_KEY);
+    Cookies.remove(LOGIN_NONCE);
+    router.push("/");
+    setAuthLoading(false);
+  };
 
-	const value: AuthContextValue = { egoJwt, setEgoJwt, authLoading, setAuthLoading, logIn, logOut };
+  const permissions = TokenUtils.getPermissionsFromToken(egoJwt);
 
-	return (
-		<AuthContext.Provider value={value}>
-			<Suspense
-				fallback={
-					<>
-						<Header />
-						<DnaLoader />
-					</>
-				}
-			>
-				{children}
-			</Suspense>
-		</AuthContext.Provider>
-	);
+  const value: AuthContextValue = {
+    egoJwt,
+    setEgoJwt,
+    authLoading,
+    setAuthLoading,
+    logIn,
+    logOut,
+    permissions,
+    TokenUtils,
+  };
+
+  return (
+    <AuthContext.Provider value={value}>
+      <Suspense
```

Comment:
there are `Suspense` boundary errors on the page and this is the only place in the code where we use suspense boundaries. I wonder if there could be a fix here?

Reply:
This is from the very early days of setting up the app + Login and definitely worth a revisit esp b/c of the errors